data/SugarBean.php

```diff
@@ -541,10 +541,11 @@ public function bean_implements($interface)
 
     /**
      * @param bool $force
+     * @throws Exception
      */
     public function populateDefaultValues($force = false)
     {
-        if (!is_array($this->field_defs)) {
+        if (!is_array($this->field_defs) && $this->module_name !== 'Audit') {
```

Comment:
I think it would actually make sense that some modules won't contain a populated ```field_defs``` so imo a more appropriate check would actually be to remove the fatal logging here and then simply move the array check to just above the ```foreach``` like so:

```php
        if (!is_array($this->field_defs)) {
            foreach ($this->field_defs as $field => $value) {
```

That way we can add more modules that don't contain ```field_defs``` in the future without having to modify this function.